ikarus/finiteelements/ferequirements.hh

```diff
@@ -88,55 +88,149 @@ namespace Ikarus {
 
 // clang-format on
 
+/**
+ * \brief Concept to check if a given type is one of the predefined affordance enums or the AffordanceCollection
+ */
+template <typename T>
+concept FEAffordance = std::is_same_v<std::remove_cvref_t<T>, ScalarAffordance> or
+                       std::is_same_v<std::remove_cvref_t<T>, VectorAffordance> or
+                       std::is_same_v<std::remove_cvref_t<T>, MatrixAffordance>;
+
 /**
  * \brief Struct representing a collection of affordances.
  */
-struct AffordanceCollectionImpl
+template <FEAffordance... Affos>
+requires(sizeof...(Affos) <= 3)
+struct AffordanceCollection : public std::tuple<Affos...>
 {
-  ScalarAffordances scalarAffordances{ScalarAffordances::noAffordance};
-  VectorAffordances vectorAffordances{VectorAffordances::noAffordance};
-  MatrixAffordances matrixAffordances{MatrixAffordances::noAffordance};
+  using Base = std::tuple<Affos...>;
+
+  AffordanceCollection() = default;
+  constexpr AffordanceCollection(Affos... affos)
+      : Base(affos...) {}
+
+  static constexpr bool hasScalarAffordance = traits::hasType<ScalarAffordance, std::tuple<Affos...>>::value;
+  static constexpr bool hasVectorAffordance = traits::hasType<VectorAffordance, std::tuple<Affos...>>::value;
+  static constexpr bool hasMatrixAffordance = traits::hasType<MatrixAffordance, std::tuple<Affos...>>::value;
+  /**
+   * \brief Check if a specific affordance is present in the requirements.
+   *
+   * This function checks if the specified affordance is present in the requirements.
+   *
+   * \tparam Affordance Type of affordance to be checked.
+   * \param affordance The affordance to be checked.
+   * \return True if the affordance is present, false otherwise.
+   */
+  template <FEAffordance Affordance>
+  constexpr bool hasAffordance(Affordance&& affordances) const {
+    if constexpr (std::is_same_v<std::remove_cvref_t<Affordance>, AffordanceCollection>)
```

Comment:
To make it a bit more readable, we could extract the Type of the forwarded Affordance
```suggestion
    template <FEAffordance Affordance_>
    constexpr bool hasAffordance(Affordance_&& affordances)
    using Affordance = std::remove_cvref_t<Affordance_>;
    if constexpr (std::is_same_v<Affordance, AffordanceCollection>)
```

Reply:
The right